### PR TITLE
fixed bugs with snow turfs,

### LINKED ISSUE #7219: Snow z-level edge
image

@DrCelt

### LINKED ISSUE #7220: SNOW IS UNPLAYABLE
JESUS FUCK SHIT FUCK DAMN THE LAG IS SO FUCKING BAD
[WAKE ME UP]

### PR DESCRIPTION
**Pull request by ghost**
fixed LAG with snow turfs fixes #7220 
randomise trees on snow turfs
now can transition on snow turfs fixes #7219 
can no longer build on transition areas

HAS BEEN 100% TESTED I GUARANTEE IT

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -181,29 +181,117 @@
...
+		for(var/atom/Obj as mob|obj|turf|area in range(1))
 			if(objects > loopsanity)	break
 			objects++
 			spawn( 0 )
-				if ((A && Obj) && A.flags & PROXMOVE)
-					A.HasProximity(Obj, 1)
-	return
+				if ((A && Obj) && Obj.flags & PROXMOVE)
+					Obj.HasProximity(A, 1)
+	// THIS IS NOW TRANSIT STUFF
```
Comment (PJB3005) [inline code/game/turfs/turf.dm:209]: One issue I can see arise from this is that people who build on the map edge are gonna be screwed.

Comment (ghost) [inline code/game/turfs/turf.dm:209]: How could they build on it in the first place?

Comment (PJB3005) [inline code/game/turfs/turf.dm:209]: What are there any checks that prevent you from building on the map edge? If so: nevermind.

Comment (ghost) [inline code/game/turfs/turf.dm:209]: there are now

**Comment by ghost** [comment]
I still need to try the snow.

**Pull request merged**